src/modules/kvs/kvs.c

```diff
@@ -203,37 +203,53 @@ static int event_subscribe (kvs_ctx_t *ctx, const char *ns)
             goto cleanup;
         }
 
+        /* On rank 0, we need to listen for all of these namespace
+         * events, all of the time.  So subscribe to them just once on
+         * rank 0. */
+        if (ctx->rank == 0) {
+            if (flux_event_subscribe (ctx->h, "kvs.setroot") < 0) {
+                flux_log_error (ctx->h, "flux_event_subscribe");
+                goto cleanup;
+            }
+
+            if (flux_event_subscribe (ctx->h, "kvs.error") < 0) {
+                flux_log_error (ctx->h, "flux_event_subscribe");
+                goto cleanup;
+            }
+
+            if (flux_event_subscribe (ctx->h, "kvs.namespace-remove") < 0) {
```

Comment:
Since the error handling for these 3 `flux_event_subscribe` calls is exactly the same, you could get another small bump in coverage by combining these conditionals into one.  Not important though, so only if you feel like it

Reply:
Good idea, actually I can combine the `flux_event_subscribe` calls above it as well.